Java/Mazes/src/Cell.java

```diff
@@ -0,0 +1,88 @@
+import java.util.ArrayList;
+
+class Cell {
+
+    private final int xCoor;
+    private final int yCoor;
+    private final ArrayList<Integer> availableDirections = new ArrayList<>();
+    private boolean visited;
+
+    /**
+     * Default Constructor
+     *
+     * @param xCoor The left/right coordinate
+     * @param yCoor The up/down coordinate
+     */
+    Cell(int xCoor, int yCoor) {
+        this.xCoor = xCoor;
+        this.yCoor = yCoor;
+    }
+
+    boolean isVisited() {
+        return !visited;
+    }
+
+    void setVisited() {
+        this.visited = true;
+    }
+
+    int getxCoor() {
+        return xCoor;
+    }
+
+    int getyCoor() {
+        return yCoor;
+    }
+
+
+
+    /**
+     * Checks that the number of available moves is greater than 0
+     *
+     * @return True is available populateAvailableDirections is greater than 0
+     */
+    boolean hasAvailableMoves() {
+        return availableDirections.size() > 0;
+    }
+
+    /**
+     * Checks the cell for available populateAvailableDirections. If the Cell is along the border, it removes the edge as an available
+     * space
+     *
+     * @param height The height of the grid
+     * @param width  The width of the grid
+     */
+    void populateAvailableDirections(int height, int width) {
+        // Check upward movement
+        if (yCoor - 1 >= 0) {
+            availableDirections.add(1);
```

Comment:
What do these magic numbers mean?

Reply:
I added comments to clarify their meaning. I added a TODO comment for replacing these with enum later on.